AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/StorageEngine.swift

```diff
@@ -415,6 +422,43 @@ final class StorageEngine: StorageEngineBehavior {
         }
     }
 
+    @available(iOS 13.0, *)
+    private func syncDeletions(of associatedModels: [Model],
+                               syncEngine: RemoteSyncEngineBehavior,
+                               completion: @escaping DataStoreCallback<Void>) {
+        guard !associatedModels.isEmpty else {
+            completion(.successfulVoid)
+            return
+        }
+        var mutationEvents: Set<Model.Identifier> = []
+        for associatedModel in associatedModels {
+            let mutationEvent: MutationEvent
+            do {
+                mutationEvent = try MutationEvent(untypedModel: associatedModel, mutationType: .delete)
+            } catch {
+                let dataStoreError = DataStoreError(error: error)
+                completion(.failure(dataStoreError))
+                return
+            }
+
+            let mutationEventCallback: DataStoreCallback<MutationEvent> = { result in
+                switch result {
```

Comment:
I note that we're invoking a completion on failure, but continuing to process events. Continuing makes sense, but it's a little weird to be invoking a completion before things have actually, you know, completed. :) Should we remember the error and then resolve the completion after all the associated models have been processed?